darts/models/forecasting/tft_model.py

```diff
@@ -0,0 +1,915 @@
+        self.prescalers_linear = {name: nn.Linear(1, self.hidden_continuous_size) for name in self.reals}
+
+        static_input_sizes = {name: self.hidden_continuous_size for name in self.static_variables}
+
+        self.static_covariates_vsn = _VariableSelectionNetwork(
+            input_sizes=static_input_sizes,
+            hidden_size=self.hidden_size,
+            input_embedding_flags={},  # this would be required for categorical inputs
+            dropout=self.dropout,
+            prescalers=self.prescalers_linear,
+            single_variable_grns={},
+            context_size=None  # no context for static variables
+        )
+
+        # variable selection for encoder and decoder
+        encoder_input_sizes = {name: self.hidden_continuous_size for name in self.encoder_variables}
+
+        decoder_input_sizes = {name: self.hidden_continuous_size for name in self.decoder_variables}
+
+        self.encoder_vsn = _VariableSelectionNetwork(
+            input_sizes=encoder_input_sizes,
+            hidden_size=self.hidden_size,
+            input_embedding_flags={},  # this would be required for categorical inputs
+            dropout=self.dropout,
+            context_size=self.hidden_size,
+            prescalers=self.prescalers_linear,
+            single_variable_grns={}
+        )
+
+        self.decoder_vsn = _VariableSelectionNetwork(
+            input_sizes=decoder_input_sizes,
+            hidden_size=self.hidden_size,
+            input_embedding_flags={},  # this would be required for categorical inputs
+            dropout=self.dropout,
+            context_size=self.hidden_size,
+            prescalers=self.prescalers_linear,
+            single_variable_grns={}
+        )
+
+        # static encoders
+        # for variable selection
+        self.static_context_grn = _GatedResidualNetwork(
+            input_size=self.hidden_size,
+            hidden_size=self.hidden_size,
+            output_size=self.hidden_size,
+            dropout=self.dropout,
+        )
+
+        # for hidden state of the lstm
+        self.static_context_initial_hidden_lstm = _GatedResidualNetwork(
+            input_size=self.hidden_size,
+            hidden_size=self.hidden_size,
+            output_size=self.hidden_size,
+            dropout=self.dropout,
+        )
+
+        # for cell state of the lstm
+        self.static_context_initial_cell_lstm = _GatedResidualNetwork(
+            input_size=self.hidden_size,
+            hidden_size=self.hidden_size,
+            output_size=self.hidden_size,
+            dropout=self.dropout,
+        )
+
+        # for post lstm static enrichment
+        self.static_context_enrichment = _GatedResidualNetwork(
+            input_size=self.hidden_size,
+            hidden_size=self.hidden_size,
+            output_size=self.hidden_size,
+            dropout=self.dropout
+        )
+
+        # lstm encoder (history) and decoder (future) for local processing
+        self.lstm_encoder = _LSTM(
```

Comment:
Do we need PTF's LSTM here? (Compared to PyTorch original LSTM).
As I understand the different is that PTF "handles 0-length sequences",  but I guess it's never the case we'll get a 0-length sequence as long as our input and output chunks are strictly positive?

Reply:
good catch!